Meadowlark.net/Meadowlark.Net.Core/Handler/FrontendFacade.cs

```diff
@@ -0,0 +1,37 @@
+using Meadowlark.Net.Core.Model;
+using Meadowlark.Net.Core.Utility;
+using static Meadowlark.Net.Core.Middleware.ApiSchemaLoadingMiddleware;
+using static Meadowlark.Net.Core.Middleware.ParsePathMiddleware;
+using static Meadowlark.Net.Core.Middleware.ValidateEndpointMiddleware;
+using static Meadowlark.Net.Core.Middleware.ValidateDocumentMiddleware;
+
+namespace Meadowlark.Net.Core;
+
+
+public static class FrontendFacade
+{
+  /**
+   * Entry point for API upsert actions
+   */
+  public static async Task<FrontendResponse> UpsertCore(FrontendRequest frontendRequest)
+  {
+    try
+    {
+      RequestModel requestModel = new(FrontendRequest: frontendRequest, PathComponents: No.PathComponents, ResourceInfo: No.ResourceInfo,
+        ApiSchema: No.ApiSchema, ResourceSchema: No.ResourceSchema);
+      MiddlewareModel middlewareModel = new(RequestModel: requestModel, FrontendResponse: null);
+
+      var (finalRequestModel, frontendResponse) = await middlewareModel
+        .SendTo(LoadApiSchema)
+        .AndThen(ParsePath)
+        .AndThen(EndpointValidation)
+        .AndThen(DocumentValidation);
```

Comment:
If these same steps are taken for all requests, we may want to implement them as [ASP.NET Core middleware](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/middleware/?view=aspnetcore-8.0). The pipeline approach above looks good, but there may be some advantages to using the middleware. One advantage might be greater familiarity for veteran C# developers.